demos/sjtwo/capture/source/main.cpp

```diff
@@ -0,0 +1,85 @@
+
+// NOTE: Connect pin 2.0 to pin 1.14 with a jumper wire to run this test.
+
+#include <cstdint>
+#include "L1_Peripheral/example.hpp"
+#include "L1_Peripheral/inactive.hpp"
+#include "L1_Peripheral/timer.hpp"
+#include "L1_Peripheral/capture.hpp"
+#include "L1_Peripheral/pwm.hpp"
+#include "L1_Peripheral/pin.hpp"
+#include "L1_Peripheral/interrupt.hpp"
+#include "L1_Peripheral/lpc40xx/pin.hpp"
+#include "L1_Peripheral/lpc40xx/timer.hpp"
+#include "L1_Peripheral/lpc40xx/capture.hpp"
+#include "L1_Peripheral/lpc40xx/pwm.hpp"
+#include "utility/log.hpp"
+#include "utility/map.hpp"
+#include "utility/time.hpp"
+#include "utility/units.hpp"
+
+sjsu::lpc40xx::Capture *capture_ref = nullptr;
+sjsu::lpc40xx::Capture::CaptureStatus_t status;
+volatile uint32_t prev_count;
+volatile uint32_t curr_count;
+volatile uint32_t delta_accumulator;
+volatile uint32_t num_samples;
+
+void CaptureISR(void)
+{
+  capture_ref->GetCaptureStatus(sjsu::Capture::kCaptureChannel0, status);
+  prev_count = curr_count;
+  curr_count = status.count;
+  delta_accumulator += (curr_count - prev_count);
+  ++num_samples;
+}
+
+int main()
+{
+  constexpr units::frequency::hertz_t kCaptureFrequency = 4_MHz;
+  constexpr units::frequency::hertz_t kGeneratorFrequency = 10_Hz;
+  constexpr float kGeneratorDutyCycle = 0.5f;
+
+  // Configure P1.14 as T2_CAP0
+  LOG_INFO("Configure P1.14 as capture input T2_CAP0");
+  sjsu::lpc40xx::Pin capture_pin(1, 14);
+  capture_pin.SetPinFunction(3);
+  capture_pin.SetAsOpenDrain(false);
```

Comment:
The capture input signals can be mapped to multiple pins, e.g. T1_CAP0 exists on P1.18 and P3.27 rather than having a 1:1 relationship where a given timer/channel maps to a single pin. One approach is that I could add a parameter to the constructor that takes an enum indicating which pin to use, but if I give a name like 'kPin1_18' then the caller could do something wacky like use that with timer 2 instead of timer 1 for the other parameter. Meaning I'd need some validation on the parameters to look for invalid combinations. Is that the right approach or not quite?

Reply:
The typical implementation is to stick with a single pin for something and allow the user to define their own structure (or copy one) and use their own pin. This way, if someone just wants to use just 1 pin to do capture, they can do this easily. If they want to use a pin we don't currently support, then they have to do the work of setting up the structure.